.github/workflows/notify-slack.yml

```diff
@@ -0,0 +1,77 @@
+---
+
+name: 'Notify slack'
+
+on:
+  workflow_call:
+    inputs:
+      application:
+        description: 'The friendly name for the application'
+        type: string
+        required: true
+
+      environment:
+        description: |
+          The environment variables to pass to the render.  Each environment
+          variable should be in format VAR=VAL with one on each line.
+          Example:
+            environment: |
+              FOO=bar
+              BAR=baz
+        type: string
+        required: false
+        default: ''
+
+      undefined:
+        description: 'If true, undefined variables will be used in templates (no error will be raised)'
+        type: string
+        required: false
+        default: 'false'
+
+      template:
+        description: 'The path to the template file to render'
+        type: string
+        required: true
+
+      version:
+        description: 'The version that is being deployed'
+        type: string
+        default: ${{ github.ref }}
+
+      type:
+        description: 'The type of action that is being performed such as build or deployment'
+        type: string
+        required: true
+
+      status:
+        description: 'The status of the deployment.  Options are started, success, and failure'
+        type: string
+        required: true
+
+    secrets:
+      slack_webhook_url:
+        description: 'The slack webhook url to post the message to'
+        required: true
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  notify:
+    name: 'Notify'
+    runs-on: ubuntu-latest
+    environment: ${{ inputs.environment }}
+    steps:
+      - name: 'Checkout'
+        uses: actions/checkout@v3
+
```

Comment:
Still unneeded.  We should not be calling any local actions here.  This action should be posting a message to slack.

This is the main workflow you are trying to make use reusable.  This should be the core logic.